applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -446,7 +446,10 @@ def _create_mechanical_solver(self):
         if analysis_type == "linear":
             mechanical_solver = self._create_linear_strategy()
         elif analysis_type == "non_linear":
-            mechanical_solver = self._create_newton_raphson_strategy()
+            if(self.settings["line_search"].GetBool() == False):
```

Comment:
@RiccardoRossi this function cannot be in the base class, it is in the `StaticMechanicalSolver`
Reasons it has to be there: (see our discussion in #850 )
- doesn't make sense in the dynamic case
- (and it does not exist in Trilinos)

We could however take the `line_search` parameter only into account if the `analysis_type` is `non_linear` (currently it is handled a bit differently, see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_static_solver.py#L102))

Reply:
well, on one side i think that this commit is out of scope here. I will remove it from this branch when i have time.

on the other hand however:
1. i agree that it only make sense if it is nonlinear
2. In the nonlinear case it does make sense (as long as the time derivatives are updated accordingly)

Reply:
Hi @RiccardoRossi,
I confused sth, so you are right.
Let me just clean this up a bit, then I am fine with your change (cleanup => move the line search struff to the base class such that it can also be used in a dynamic analysis)